### PR TITLE
Transition to standard smart pointers, part 11

### PR DESCRIPTION
**Pull request by SergioRAgostinho**
Partially addresses #2792.

### CODE REVIEW

File: sample_consensus/include/pcl/sample_consensus/sac.h [hunk 1]
```diff
@@ -66,6 +66,8 @@ namespace pcl
       using Ptr = boost::shared_ptr<SampleConsensus<T> >;
       using ConstPtr = boost::shared_ptr<const SampleConsensus<T> >;
 
+      using IndicesPtr = boost::shared_ptr<std::vector<int>>;
```
Comment (taketwo) [inline sample_consensus/include/pcl/sample_consensus/sac.h]: Is there a reason not to use our global `IndicesPtr` type here?

https://github.com/PointCloudLibrary/pcl/blob/65ecbd855ea96ffecac4fb32b4ab9224e1bc225c/common/include/pcl/pcl_base.h#L61

File: sample_consensus/include/pcl/sample_consensus/sac_model.h [hunk 2]
```diff
@@ -75,6 +75,8 @@ namespace pcl
       using Ptr = boost::shared_ptr<SampleConsensusModel<PointT> >;
       using ConstPtr = boost::shared_ptr<const SampleConsensusModel<PointT> >;
 
+      using IndicesPtr = boost::shared_ptr<std::vector<int>>;
```
Comment (taketwo) [inline sample_consensus/include/pcl/sample_consensus/sac_model.h]: Also here

File: sample_consensus/include/pcl/sample_consensus/sac_model_registration.h [hunk 3]
```diff
@@ -217,6 +218,7 @@ namespace pcl
     protected:
       using SampleConsensusModel<PointT>::sample_size_;
       using SampleConsensusModel<PointT>::model_size_;
+      using typename SampleConsensusModel<PointT>::IndicesPtr;
```
Comment (taketwo) [inline sample_consensus/include/pcl/sample_consensus/sac_model_registration.h]: And here

File: sample_consensus/include/pcl/sample_consensus/sac.h (hunk 1 above)
Comment (SergioRAgostinho) [reply sample_consensus/include/pcl/sample_consensus/sac.h]: Nice catch. I didn't check if the base class had anything!

**Pull request merged**